src/ImageSharp/Formats/Tiff/Compression/Decompressors/JpegTiffCompression.cs

```diff
@@ -55,17 +54,43 @@ protected override void Decompress(BufferedReadStream stream, int byteCount, int
             {
                 using var jpegDecoder = new JpegDecoderCore(this.configuration, new JpegDecoder());
 
-                // If the PhotometricInterpretation is YCbCr we explicitly assume the JPEG data is in RGB color space.
-                // There seems no other way to determine that the JPEG data is RGB colorspace (no APP14 marker, componentId's are not RGB).
-                using SpectralConverter<Rgb24> spectralConverter = this.photometricInterpretation == TiffPhotometricInterpretation.YCbCr ?
-                    new RgbJpegSpectralConverter<Rgb24>(this.configuration) : new SpectralConverter<Rgb24>(this.configuration);
-                var scanDecoder = new HuffmanScanDecoder(stream, spectralConverter, CancellationToken.None);
-                jpegDecoder.LoadTables(this.jpegTables, scanDecoder);
-                scanDecoder.ResetInterval = 0;
-                jpegDecoder.ParseStream(stream, scanDecoder, CancellationToken.None);
+                switch (this.photometricInterpretation)
+                {
+                    case TiffPhotometricInterpretation.BlackIsZero:
+                    case TiffPhotometricInterpretation.WhiteIsZero:
+                    {
+                        using SpectralConverter<L8> spectralConverterGray = new GrayJpegSpectralConverter<L8>(this.configuration);
+                        var scanDecoderGray = new HuffmanScanDecoder(stream, spectralConverterGray, CancellationToken.None);
+                        jpegDecoder.LoadTables(this.jpegTables, scanDecoderGray);
+                        scanDecoderGray.ResetInterval = 0;
```

Comment:
What is the point of zeroing this field? It should be zero by default if there is no RST marker.

Reply:
Yes you are right, that is not needed. Changed with 2fbf5664